benchmarks/triton_kernels_benchmark/benchmark_utils.py

```diff
@@ -0,0 +1,210 @@
+        ]
+        return "\n".join(str_repr)
+
+    def run(self, dry_run: bool, args: MarkArgs):
+        print(str(self))
+        if dry_run:
+            return
+        # FIXME: add run summary - total timings, etc
+        print(f"Running {self.key}")
+        self._get_benchmark().run(show_plots=False, print_data=True, mark_args=args)
+
+
+@dataclass
+class BenchmarkConfigs:
+    configs: List[BenchmarkConfig]
+    dry_run: bool
+    args: MarkArgs
+
+    _benchmark_config_templates: ClassVar[List[BenchmarkConfig]] = [
+        # To filter the providers list you can run console entry point with multiple `--provider <supported_provider>` options
+        BenchmarkConfig(
+            key="fused_softmax",
+            get_benchmark=fused_softmax.get_benchmark,
+            config_opts={},
+            category=BenchmarkCategory.CORE,
+        ),
+        BenchmarkConfig(
+            key="gemm",
+            get_benchmark=gemm_benchmark.get_benchmark,
+            config_opts={},
+            category=BenchmarkCategory.CORE,
+        ),
+        BenchmarkConfig(
+            key="gemm_bt",
+            get_benchmark=gemm_benchmark.get_benchmark,
+            config_opts={"transpose_b": True},
+            category=BenchmarkCategory.CORE,
+        ),
+        BenchmarkConfig(
+            key="flash_attention",
+            get_benchmark=flash_attention_benchmark.get_benchmark,
+            config_opts={"fa_kernel_mode": "fwd"},
+            category=BenchmarkCategory.CORE,
+        ),
+    ]
+
+    @classmethod
+    def _get_configs(cls, suite: str, categories_filter: List[str]) -> List[BenchmarkConfig]:
+        selected_config = None
+        all_configs = cls._benchmark_config_templates
+        config_keys = []
+        for config in all_configs:
+            key = config.key
+            if key == "all":
+                raise AssertionError("`all` can't be a config name key")
+            if key in config_keys:
+                raise ValueError(f"Duplicate config name key - {key}")
+            config_keys.append(key)
+            if key == suite:
+                selected_config = config
+        if selected_config and selected_config.category.value not in categories_filter:
+            raise AssertionError(
+                f"Selected config {selected_config.key} category {selected_config.category} is not in the categories filter"
+            )
+        if selected_config:
+            return [selected_config]
+        if suite == "all":
+            return [config for config in all_configs if config.category.value in categories_filter]
+        raise NotImplementedError(
+            f"Config {suite} is unknown. Choose `all` or one of the following: " + ", ".join(config_keys), )
+
+    def run(self):
+        start = time.perf_counter()
+        for config in self.configs:
+            config.run(args=self.args, dry_run=self.dry_run)
+        end = time.perf_counter()
+        print(f"Total run time: {round(end-start, 3)}")
```

Comment:
For what? We collect timings inside the run function.
```suggestion
        for config in self.configs:
            config.run(args=self.args, dry_run=self.dry_run)
```

Reply:
The intended use case is too give command line user an easy indication how much time it took to run benchmarks configs with the filters provided. This time measurement should have nothing to do with the individual timings for benchmark run per provider and config.